File: docs/articles_en/assets/snippets/compile_model_npu.cpp
#include <openvino/runtime/core.hpp>

int main() {
{
    //! [compile_model_default_npu]
    ov::Core core;
    auto model = core.read_model("model.xml");
    auto compiled_model = core.compile_model(model, "NPU");
    //! [compile_model_default_npu]
}
    return 0;
}

File: docs/articles_en/assets/snippets/compile_model_npu.py
# Copyright (C) 2018-2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

import openvino as ov
from snippets import get_model


def main():
    model = get_model()

    core = ov.Core()
    if "NPU" not in core.available_devices:
        return 0

    #! [compile_model_default_npu]
    core = ov.Core()
    compiled_model = core.compile_model(model, "NPU")
    #! [compile_model_default_npu]
